tests/test-lexer.cpp

```diff
@@ -118,7 +116,7 @@ TEST_CASE("Test the Schema class", "[Schema]") {
         auto* regex_ast_capture
                 = dynamic_cast<RegexASTCaptureByte const*>(regex_ast_cat_ptr->get_right());
         REQUIRE(nullptr != regex_ast_capture);
-        REQUIRE("uID" == regex_ast_capture->get_group_name());
+        REQUIRE("uID" == string{regex_ast_capture->get_group_name()});
```

Comment:
We don't need to construct a `string`. Instead, we can directly compare a string view with a static string:
```suggestion
        REQUIRE("uID" == regex_ast_capture->get_group_name());
```
We should apply this to `test-tag.cpp` as well

Reply:
This doesn't work because Catch2 is complaining about string_view comparisons. Not sure if I'm just doing something wrong, or there is a better workaround.

Reply:
Changed.